### PR TITLE
    Validation/RecoEgamma : Fix clang warning [-Wtautological-constant-out-of-range-compare]

### PR DESCRIPTION
**Pull request by gartung**
Validation/RecoEgamma : Fix clang warning [-Wtautological-constant-out-of-range-compare]

    warning: comparison of constant 140 with expression of type 'reco::GsfElectron::Classification' is always true [-Wtautological-constant-out-of-range-compare]
    classification()!= 140);
    ~~~~~~~~~~~~~~~~^  ~~~

    Store return value on ClassificationVariables::classification() in a temporary variable and then compare to the "magic numbers". This is obviously an unused algor

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones

File: Validation/RecoEgamma/plugins/ElectronMcFakeValidator.cc [hunk 1]
```diff
@@ -1061,8 +1061,9 @@ void ElectronMcFakeValidator::analyze( const edm::Event & iEvent, const edm::Eve
         h1_scl_Et_->Fill(sclRef->energy()*(Rt/R));
         h2_scl_EtVsEta_->Fill(sclRef->eta(),sclRef->energy()*(Rt/R));
         h2_scl_EtVsPhi_->Fill(sclRef->phi(),sclRef->energy()*(Rt/R));
-        if (bestGsfElectron.classification() < 100)  h1_scl_EoEmatchingObject_barrel->Fill(sclRef->energy()/moIter->energy());
-        if (bestGsfElectron.classification() >= 100)  h1_scl_EoEmatchingObject_endcaps->Fill(sclRef->energy()/moIter->energy());
+        int c = bestGsfElectron.classification();
+        if ( c < 100)  h1_scl_EoEmatchingObject_barrel->Fill(sclRef->energy()/moIter->energy());
```
Comment (Dr15Jones) [inline Validation/RecoEgamma/plugins/ElectronMcFakeValidator.cc]: I think you want to call 
```
if(bestGsfElectron.isEB())  { ...
} else {
...
}
```

**Comment by Dr15Jones** [comment]
please test

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**